src/overlays/actors/ovl_Dm_Sa/z_dm_sa.c

```diff
@@ -19,28 +20,102 @@ const ActorInit Dm_Sa_InitVars = {
     (ActorFunc)DmSa_Init,
     (ActorFunc)DmSa_Destroy,
     (ActorFunc)DmSa_Update,
-    (ActorFunc)DmSa_Draw
+    (ActorFunc)DmSa_Draw,
 };
-*/
 
-#pragma GLOBAL_ASM("./asm/non_matchings/overlays/ovl_Dm_Sa_0x80A2E960/func_80A2E960.asm")
+extern SkeletonHeader D_06013328;
+extern AnimationHeader D_0600CC94;
+
+static ActorAnimationEntry D_80A2ED00[] = { { &D_0600CC94, 1.0f, 0, -1.0f, 0, 0 } };
+
+void func_80A2E960(SkelAnime* arg0, ActorAnimationEntry* animations, u16 index) {
+    f32 frameCount;
+    animations += index;
+
+    if (animations->frameCount < 0.0f) {
+        frameCount = (f32)SkelAnime_GetFrameCount(animations->animation);
+    } else {
+        frameCount = animations->frameCount;
+    }
+    SkelAnime_ChangeAnim(arg0, animations->animation, animations->playSpeed, animations->startFrame, frameCount,
+                         animations->mode, animations->morphFrames);
+}
+
+void DmSa_Init(Actor* thisx, GlobalContext* globalCtx) {
+    DmSa* this = THIS;
+
+    this->unk2E0 = 0;
+    this->alpha = 0xFF;
+    this->actor.targetArrowOffset = 3000.0f;
+    Actor_SetDrawParams(&this->actor.shape, 0.0f, func_800B3FC0, 24.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_06013328, NULL, 0, 0, 0);
+    func_80A2E960(&this->skelAnime, &D_80A2ED00, 0);
+    Actor_SetScale(&this->actor, 0.01f);
+    this->actionFunc = DmSa_DoNothing;
+}
+
+void DmSa_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
+
+void DmSa_DoNothing(DmSa* this, GlobalContext* globalCtx) {
+}
+
+void DmSa_Update(Actor* thisx, GlobalContext* globalCtx) {
+    DmSa* this = THIS;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    this->alpha += 0;
+    this->actionFunc(this, globalCtx);
+}
+
+s32 func_80A2EB10(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+    return 0;
+}
+
+void func_80A2EB2C(s32 arg0, s32 arg1, s32 arg2, s32 arg3) {
+}
+
+void func_80A2EB44(s32 arg0, s32 arg1, s32 arg2) {
+}
+
+Gfx* func_80A2EB58(GraphicsContext* gfxCtx, u32 alpha) {
+    Gfx* dList;
+    Gfx* dListHead;
+
+    dList = dListHead = GRAPH_ALLOC(gfxCtx, sizeof(Gfx) * 2);
```

Comment:
I saw the discussion about this bug, so I'd put
```suggestion
    dList = dListHead = GRAPH_ALLOC(gfxCtx, sizeof(Gfx) * 2); //! @bug this does not allocate enough for 3 Gfx commands
```